### ISSUE TITLE
Detailed display of taxes on the order view

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#20817]
#### Describe the bug

On BO order edit, the merchant is unable to determine the amount of tax applied to product and the whole order.
It's important information.

#### Desired solution

Total tax should be displayed on the screen.
Subtotals should be with and without tax.
Total should be with and without tax.

Something like this maybe?
![amounts](https://user-images.githubusercontent.com/6097524/91965685-20e00280-ed11-11ea-8a85-ae683b07b026.PNG)

#### Additional information

* PrestaShop version: 1.7.7 beta 2
* PHP version: 7.2

**Comment by sarahdib** [comment]
@PrestaShop/prestashop-product-team what do you think about this improvement ?

**Comment by LouiseBonnard** [comment]
Or add an option to let the user determines if he/she wants to have this information or not? @MatShir, I let you have a look at those suggestions. ;-)

**Comment by MatShir** [comment]
**Edit**: Indeed, we could enhance the tax display. I can imagine why it is important to have a detailed of the taxes, but it is nuisance for daily life? Do your merchants check every time the tax or more at the end of the month for example?

**Comment by Hlavtox** [comment]
@MatShir @LouiseBonnard Not daily, but imagine you are on a call with another business customer, a VAT payer. You deal some stuff for example, you make him an order manually. Bla bla bla. You are checking his order and now following happens:

He asks: _"What's the total without tax?"_ (He doesn't care about tax, he will get it back from the state)
You say: _"Oooh let me take out my calculator."_, because you don't see it there! :-D

**To sum it up - general customer cares about the price with tax, business customers care about the price without tax.**

**Comment by MatShir** [comment]
related to:
- [ ] #20821 - Order create - Confusing display of amounts, taxes and currencies
- [ ]  #15954 - Clearify user groups and tax included/excluded display
Maybe work can be done for all this issues

**Comment by hibatallahAouadni** [comment]
@MatShir isn't similar to this one #12746? could you please check :pray:

**Comment by tswfi** [comment]
Finnish tax authority defines pretty strict rules on what should be shown in the receipts for customers:
https://www.vero.fi/en/businesses-and-corporations/about-corporate-taxes/tax-auditing-and-shadow-economy/kuitinantovelvollisuus_harmaan_talouden/play_fair_dont_forget_the_receip/

The example receipt is for a POS system but the same rule applies to order confirmation and other documents provided.

I think this kind of tax summary would be beneficial for everybody.

**Comment by Hlavtox** [comment]
@tswfi Lets implement it for 8.0?

**Comment by tswfi** [comment]
> @tswfi Lets implement it for 8.0?

can you find more hours to my days somewhere :D

but yes, this is one the things that are on my backlog and needs to be addressed as soon as I have time...

**Comment by OrionCodeDev** [comment]
Any update?

**Comment by boo-code** [comment]
The first ask here - "Total tax should be displayed on the screen" - was already satisfied when this was filed. The order view summary has carried a `Taxes` figure since at least 1.7.7.0, the version in the report. Read from the release tags with `orderProductsTotal` taken from the same blob as a control, so an unreadable ref cannot be mistaken for an absent row:

| ref | `orderTaxesTotal` row |
| --- | --- |
| `1.7.7.0` | present |
| `1.7.7.8` | present |
| `1.7.8.0` | present |
| `8.1.0` | present |
| `8.2.0` | present |
| `9.0.0` | present |
| `9.1.5` | present |
| `9.2.x` | present |

What the page does today, from `GetOrderForViewingHandler::getOrderPrices()`:

```php
$isTaxExcluded = !$this->isTaxIncludedInOrder($order);

if ($isTaxExcluded) {
    $productsPrice = (float) $order->total_products;
    $discountsAmount = (float) $order->total_discounts_tax_excl;
    // wrapping, shipping likewise tax excluded
} else {
    $productsPrice = (float) $order->total_products_wt;
    // ... tax included
}
$totalAmount = (float) $order->total_paid_tax_incl;
$taxesAmount = $order->total_paid_tax_incl - $order->total_paid_tax_excl;
```

The subtotals follow the customer group's tax method. The total does not: it is unconditionally tax included. For a tax-excluded group that is not an oversight, it is what makes the column read as an invoice - net lines, then tax, then gross total. Checked against a real order: products 12.90 + shipping 7.00 + tax 1.40 = 21.30, which is exactly its `total_paid_tax_incl`.

That matters for the shape of any fix. Putting a tax-excluded total on the same row gives the column two totals that disagree with each other, so "show both" is a layout decision rather than an extra value.

The concrete blocker for whoever implements it: `OrderPricesForViewing` carries exactly one basis per line - seven amounts, seven getters, no excluded/included pair anywhere. Both bases means widening that query result and everything reading it, not only the Twig block.

The shape itself is still open, and this thread holds four different ones: always visible, a merchant preference, an excluded total beside the included one, or the statutory receipt layout the Finnish rules describe.
